crawlerr/models/source_models.py

```
@@ -0,0 +1,154 @@
+    limit: int = Field(default=100, ge=1, le=1000)
+    offset: int = Field(default=0, ge=0)
+    
+    def matches_source(self, source: SourceInfo) -> bool:
+        """Check if a source matches the filter criteria."""
+        # Type filter
+        if self.source_types and source.source_type not in self.source_types:
+            return False
+        
+        # Domain filter
+        if self.domains and source.metadata.domain:
+            if not any(domain in source.metadata.domain for domain in self.domains):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)